src/Sulu/Bundle/DocumentManagerBundle/Bridge/DocumentInspector.php

```diff
@@ -130,6 +131,23 @@ public function getMetadata($document)
     }
 
     /**
+     * Returns the first ancestor-or-self of the given document which is published and therefore has an
+     * assigned resource locator. If all ancestor documents are unpublished, the root document is returned.
+     *
+     * @param object    $document
+     *
+     * @return object
+     */
+    public function getPublishedAncestorOrSelf($document)
```

Comment:
I think I wouldn't add such a specific method to the `DocumentInspector`... Isn't there a better was to do this? E.g. moving the logic of finding the correct document to generate the route in the `RlpStrategy`?